### PR TITLE
fix the bug of named_parameters

### PR DESCRIPTION
**Pull request by mileyan**
When I use the named_parametes to modify the lr and weight decay, I will face a bug. Because the value of the named_parameters return is  torch.nn.paramter.Parameter, not a generator of the Parameter.
the error information:
https://discuss.pytorch.org/t/problem-on-different-learning-rate-and-weight-decay-in-different-layers/3619

### CODE REVIEW

File: torch/optim/optimizer.py [hunk 1]
```diff
@@ -33,7 +33,10 @@ def __init__(self, params, defaults):
 
         param_set = set()
         for group in self.param_groups:
-            group['params'] = list(group['params'])
+            if isinstance(group['params'], torch.nn.paramter.Parameter):
```
Comment (apaszke) [inline torch/optim/optimizer.py]: You should check for an autograd Variable, not a Parameter

**Review by apaszke — CHANGES_REQUESTED** [review]
Can you also add a test case for this?

**Comment by apaszke** [comment]
It's still missing the test case. I'll merge it once it's added.

**Comment by soumith** [comment]
i've added a test for this and merged your PR into master. Thank you.

**Comment by soumith** [comment]
merged via https://github.com/pytorch/pytorch/commit/a76098ac1532d5e9ee24b4776258ae731627f8e3
